components/textfield/index.css

```diff
@@ -10,116 +10,445 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+@media (forced-colors: active) {
+  .spectrum-Textfield {
+    --spectrum-textfield-border-color-hover: Highlight;
+    --spectrum-textfield-border-color-keyboard-focus: Highlight;
+
+    --spectrum-textfield-border-color-invalid-default: Highlight;
+    --spectrum-textfield-border-color-invalid-hover: Highlight;
+    --spectrum-textfield-border-color-invalid-keyboard-focus: Highlight;
+
+    --spectrum-textfield-text-color-default: GrayText;
+    --spectrum-textfield-text-color-hover: GrayText;
+    --spectrum-textfield-text-color-keyboard-focus: GrayText;
+    --spectrum-textfield-text-color-disabled: GrayText;
+    --spectrum-textfield-text-color-readonly: CanvasText;
+
+    --spectrum-textfield-focus-indicator-color: Highlight;
+
+    &.is-focused,
+    &.is-keyboardFocused {
+      .spectrum-Textfield-input-wrapper {
+        /* focus indicator is focused state */
+        &::after {
+          forced-color-adjust: none;
+        }
+      }
+    }
   }
+}
+
+
+/********* TEXT FIELD and TEXT AREA Outer Wrapper *********/
+.spectrum-Textfield {
+  position: relative;
+  outline: none;
+  margin: auto;
+
+  /*** ↓ Browser Mitigations for Text Field ↓ ***/
+  /* Edge - Use padding instead of text-indent because text-indent does not left align the text  */
+  text-indent: 0;
+
+  /* Edge - Show the overflow for input. */
+  overflow: visible;
+
+  /* Firefox and Safari - Remove the margin for input. */
+  margin: 0;
 
-  /* todo: unclear how to use --spectrum-textfield-m-text-padding-top and friends */
-  --spectrum-textfield-texticon-padding-top: 3px;
-  --spectrum-textfield-texticon-padding-bottom: 5px;
+  /* Firefox -
+    Removes the native spin buttons in Firefox; -moz-appearance: none results in spinners.
+    This has to come after -webkit-appearance or it gets overridden (#214)
+    Details: http://stackoverflow.com/questions/23372903/hide-spinner-in-input-number-firefox-29
+
+    Sets the opacity to 1 as normalize.css sets an opacity to placeholders
+    Details: https://github.com/csstools/normalize.css/blob/main/normalize.css#L297
+  */
+  -moz-appearance: textfield;
+
+  /* All browsers - Change the input font styles */
+  text-overflow: ellipsis;
+  -webkit-appearance: none;
 
-  /* todo: DNA uses incorrect font family "Adobe Clean" */
-  --spectrum-textfield-texticon-text-font-family: var(--spectrum-alias-body-text-font-family);
+  /* Grid layout for child components */
+  display: inline-grid;
+  grid-template-columns: auto auto auto;
+  grid-template-rows: auto auto;
+}
 
-  /* todo: add to DNA */
-  --spectrum-textfield-texticon-icon-gap: var(--spectrum-global-dimension-size-65);
-  --spectrum-textfield-quiet-texticon-icon-gap: var(--spectrum-global-dimension-size-75);
+/********* Child Component - Label *********/
+.spectrum-FieldLabel {
+  .spectrum-Textfield & {
+    grid-row: 1;
+    grid-column: 1 / span 2;
+    margin-block-end: var(--mod-textfield-label-spacing-block, var(--spectrum-textfield-label-spacing-block));
```

Comment:
I think the label, character count, and help text elements will need to push away a bit (the border radius size or maybe half the border radius?) from the input field sides. Due to the rounded corners, even though they're perfectly aligned, they look a bit offset to the user:

<img width="271" alt="Screenshot 2022-12-02 at 4 31 23 PM" src="https://user-images.githubusercontent.com/1840295/205392743-fec0dec7-9992-4108-bcc0-e7be224b0f37.png">

Reply:
I added half the border radius as some additional padding for those elements. Let me know what you think!